free-programming-books-es.md

```diff
@@ -115,194 +122,198 @@
+* [¡Aprende Haskell por el bien de todos!](http://aprendehaskell.es/main.html) (HTML)
+* [Piensa en Haskell (ejercicios de programación funcional)](http://www.cs.us.es/~jalonso/publicaciones/Piensa_en_Haskell.pdf) (PDF)
+
+
+### HTML / CSS
+
+* [CSS avanzado](http://librosweb.es/libro/css_avanzado) Javier Eguíluz (HTML) [(PDF)](https://openlibra.com/es/book/download/css-avanzado)
+* [CSS3 y JavaScript avanzado](https://openlibra.com/es/book/download/css3-y-javascript-avanzado) (PDF)
+* [Diseño de Interfaces Web](http://interfacesweb.github.io/unidades/) (HTML)
+* [Estructura con CSS](http://es.learnlayout.com) (HTML)
+* [Guía Completa de CSS3](https://openlibra.com/es/book/download/guia-completa-de-css3) - Antonio Navajas (PDF)
+* [HTML5](http://www.arkaitzgarro.com/html5) - Arkaitz Garro (HTML) [(PDF)](https://openlibra.com/es/book/download/html5)
+* [Introducción a CSS](http://librosweb.es/libro/css/) - Javier Eguíluz (HTML) [(PDF)](https://openlibra.com/es/book/download/introduccion-a-css)
 
 
 ### Java
 
-* [Aprendiendo Java](http://www.decom-uv.cl/~roberto/apuntes/java/AprendiendoJava.pdf) (PDF)
-* [Arquitectura Java Sólida](http://www.arquitecturajava.com)
+* [Aprendiendo Java y POO (2008)](http://www.decom-uv.cl/~roberto/apuntes/java/AprendiendoJava.pdf) (PDF)
 * [Desarrollo de proyectos informáticos con Java](http://www3.uji.es/~belfern/libroJava.pdf) (PDF)
-* [Notas de Introducción al Lenguaje de Programación Java](http://lya.fciencias.unam.mx/jloa/publicaciones/introduccionJava.pdf) - Jorge L. Ortega Arjona, UNAM (PDF)
-* [OCPJP6 Resumen Español](https://github.com/PabloReyes/ocpjp-resumen-espanol) - Pablo Reyes Almagro (PDF)
-* [Pensando la computación como un científico (con Java)](http://www.ungs.edu.ar/areas/publicaciones/476/pensando-la-computacion-como-un-cientifico.html)
-* [PlugIn Tapestry: Desarrollo de aplicaciones y páginas web con Apache Tapestry](http://picodotdev.github.io/blog-bitix/2014/02/libro-sobre-desarrollo-de-aplicaciones-con-apache-tapestry/) ([PDF](http://picodotdev.github.io/blog-bitix/assets/custom/PlugInTapestry.pdf)) ([EPUB](http://picodotdev.github.io/blog-bitix/assets/custom/PlugInTapestry.epub)) ([MOBI](http://picodotdev.github.io/blog-bitix/assets/custom/PlugInTapestry.mobi))
-* [Programación Orientada a Objetos en Java](http://fcasua.contad.unam.mx/apuntes/interiores/docs/98/opt/java.pdf) (PDF)
+* [Ejercicios de Programación en Java](https://www.arkaitzgarro.com/java) - F.M. Pérez Montes (PDF)
+* [Java](https://www.arkaitzgarro.com/java/) (HTML)
+* [Notas de Introducción al Lenguaje de Programación Java (2004)](http://lya.fciencias.unam.mx/jloa/publicaciones/introduccionJava.pdf) - Jorge L. Ortega Arjona (PDF)
+* [Prácticas d Java (2009)](https://openlibra.com/es/book/download/practicas-de-java) (PDF)
+* [Pensando la computación como un científico (con Java)](http://www.ungs.edu.ar/cm/uploaded_files/publicaciones/476_cid03-Pensar%20la%20computacion.pdf) (PDF)
```

Comment:
out of order

Reply:
Why? I checked, and the links are working.
Because they have a slow charge? What do you mean with out of service?